lib/util/git/index.ts

```diff
@@ -42,6 +42,90 @@ export { setPrivateKey } from './private-key';
 // TODO: fix upstream types https://github.com/steveukx/git-js/issues/704
 const ResetMode = (simpleGit.default as any).ResetMode as typeof _ResetMode;
 
+// A generic wrapper for simpleGit.* calls to make them more fault-tolerant
+async function gitRetry(
+  retryCount: number,
+  delaySeconds: number,
+  delayFactor: number,
+  gitFuncName: string,
+  gitFunc: (...args) => Promise<any>,
+  args: any[] | undefined
+): Promise<any> {
+  logger.debug(
+    { retryCount, delaySeconds, delayFactor, gitFuncName, args },
+    'gitRetry'
+  );
+  let round: number = 0;
+  while (round <= retryCount) {
+    logger.debug({ round }, 'gitRetry round');
+    try {
+      if (args === undefined) {
+        return await gitFunc.call(this as SimpleGit);
+      }
+      return await gitFunc.call(this as SimpleGit, args);
+    } catch (err) {
+      logger.debug({ err }, `${gitFuncName} thrown`);
+      try {
+        checkForPlatformFailure(err);
+        throw err;
+      } catch (errChecked) {
+        if (errChecked instanceof ExternalHostError) {
```

Comment:
This will never be true, as simple git won't throw our exception.

Reply:
I use callback now, so the exception is thrown